packages/flutter/lib/src/gestures/tap.dart

```diff
@@ -98,6 +98,197 @@ typedef GestureTapCallback = void Function();
+  /// the tap sequence.
+  ///
+  /// If the gesture doesn't win the arena, [handleTapUp] is called next.
+  /// Otherwise, [handleTapCancel] is called next.
+  @protected
+  void handleTapDown({ PointerDownEvent down });
+
+  /// A pointer that will trigger a tap has stopped contacting the screen.
+  ///
+  /// This triggers on the up event, if the gesture wins the arena with it
+  /// or has previously won.
+  ///
+  /// The parameter `down` is the down event of the primary pointer that started
+  /// the tap sequence, and `up` is the up event that ended the tap sequence.
+  ///
+  /// If the gesture doesn't win the arena, [handleTapCancel] is called
+  /// instead.
+  @protected
+  void handleTapUp({ PointerDownEvent down, PointerUpEvent up });
+
+  /// The pointer that previously triggered [handleTapDown] will not end up
+  /// causing a tap.
+  ///
+  /// This triggers once the gesture loses the arena, if [handleTapDown] has
+  /// been previously triggered.
+  ///
+  /// The parameter `down` is the down event of the primary pointer that started
+  /// the tap sequence; `cancel` is the cancel event, which might be null;
+  /// `reason` is a short description of the cause if `cancel` is null, which
+  /// can be `"forced"` if other gestures won the arena, or `"spontaneous"`
+  /// otherwise.
+  ///
+  /// If the gesture wins the arena, [onSecondaryTapUp] is called instead.
+  @protected
+  void handleTapCancel({ PointerDownEvent down, PointerCancelEvent cancel, String reason });
+
+  @override
+  void addAllowedPointer(PointerDownEvent event) {
+    if (state == GestureRecognizerState.ready) {
+      // `_down` must be assigned in this method instead of `handlePrimaryPointer`,
+      // because `acceptGesture` might be called before `handlePrimaryPointer`,
+      // which relies on `_down` to call `_delegate.onDown`.
+      _down = event;
+    }
+    super.addAllowedPointer(event);
+  }
+
+  @override
+  void handlePrimaryPointer(PointerEvent event) {
+    if (event is PointerUpEvent) {
+      _up = event;
+      _checkUp();
+    } else if (event is PointerCancelEvent) {
+      resolve(GestureDisposition.rejected);
+      if (_sentTapDown) {
+        _checkCancel(event, '');
+      }
+      _reset();
+    } else if (event.buttons != _down.buttons) {
+      resolve(GestureDisposition.rejected);
+      stopTrackingPointer(primaryPointer);
+    }
+  }
+
+  @override
+  void resolve(GestureDisposition disposition) {
+    if (_wonArenaForPrimaryPointer && disposition == GestureDisposition.rejected) {
+      // This can happen if the gesture has been canceled. For example, when
+      // the pointer has exceeded the touch slop, the buttons have been changed,
+      // or if the recognizer is disposed.
+      assert(_sentTapDown);
+      _checkCancel(null, 'spontaneous');
+      _reset();
+    }
+    super.resolve(disposition);
+  }
+
+  @override
+  void didExceedDeadlineWithEvent(PointerDownEvent event) {
```

Comment:
If you don't care about the event passed in, should you just override didExceedDeadline?